synth/src/version.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::Result;
+use core::option::Option;
+use core::option::Option::{None, Some};
+use core::result::Result::Ok;
+use core::time::Duration;
+use semver::Version;
+use serde_json::map::Map;
+use serde_json::value::Value;
+use reqwest::header::USER_AGENT;
+use crate::cli::config;
+
+
+pub fn version() -> String {
+    env!("CARGO_PKG_VERSION").to_string()
+}
+
+fn version_semver() -> Result<Version> {
+    Version::parse(&version())
+        .map_err(|e| anyhow!("failed to parse current version semver with error: {}", e))
+}
+
+fn has_notified_for_version(version: Version) -> bool {
+    // If no versions have been seen yet, default to empty map.
+    let mut seen_versions = config::get_seen_versions().unwrap_or_default();
+    let version_as_string = version.to_string();
+
+    // If the set did not have this value present, true is returned.
+    let has_notified = !seen_versions.insert(version_as_string);
```

Comment:
Why do we need to save all seen versions here? Wouldn't the latest seen version be enough?

Reply:
Good question. I was thinking in case we roll back latest.